include/oneapi/dpl/experimental/kt/esimd_radix_sort_dispatchers.h

```diff
@@ -0,0 +1,323 @@
+    }
+
+    __event_chain = __q.submit(
+        [__event_chain, __p_tmp_mem, __q](sycl::handler& __cgh)
+        {
+            __cgh.depends_on(__event_chain);
+            __cgh.host_task([=]() { sycl::free(__p_tmp_mem, __q); });
+        });
+
+    return __event_chain;
+}
+
+template <typename _KernelName, bool __is_ascending, ::std::uint8_t __radix_bits, ::std::uint16_t __data_per_work_item,
+          ::std::uint16_t __work_group_size, typename _KeysRng, typename _ValsRng>
+sycl::event
+__onesweep_by_key(sycl::queue __q, _KeysRng&& __keys_rng, _ValsRng&& __vals_rng, ::std::size_t __n)
+{
+    using _KeyT = oneapi::dpl::__internal::__value_t<_KeysRng>;
+    using _ValT = oneapi::dpl::__internal::__value_t<_ValsRng>;
+
+    using _EsimdRadixSortHistogram = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<
+        __esimd_radix_sort_onesweep_histogram<_KernelName>>;
+    using _EsimdRadixSortScan = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<
+        __esimd_radix_sort_onesweep_scan<_KernelName>>;
+    using _EsimdRadixSortSweepEven = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<
+        __esimd_radix_sort_onesweep_even_by_key<_KernelName>>;
+    using _EsimdRadixSortSweepOdd = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<
+        __esimd_radix_sort_onesweep_odd_by_key<_KernelName>>;
+    using _EsimdRadixSortCopyback = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<
+        __esimd_radix_sort_onesweep_copyback_by_key<_KernelName>>;
+
+    using _GlobalHistT = ::std::uint32_t;
+    constexpr ::std::uint32_t __bin_count = 1 << __radix_bits;
+
+    // TODO: consider adding a more versatile API, e.g. passing special kernel_config parameters for histogram computation
+    constexpr ::std::uint32_t __hist_work_group_count = 64;
+    constexpr ::std::uint32_t __hist_work_group_size = 64;
+
+    const ::std::uint32_t __sweep_work_group_count = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __work_group_size * __data_per_work_item);
+    constexpr ::std::uint32_t __bit_count = sizeof(_KeyT) * 8;
+    constexpr ::std::uint32_t __stage_count = oneapi::dpl::__internal::__dpl_ceiling_div(__bit_count, __radix_bits);
+
+    constexpr ::std::uint32_t __hist_mem_size = __bin_count * sizeof(_GlobalHistT);
+    constexpr ::std::uint32_t __global_hist_mem_size = __stage_count * __hist_mem_size;
+    const ::std::uint32_t __group_hists_mem_size = __sweep_work_group_count * __stage_count * __hist_mem_size;
+    const ::std::uint32_t __tmp_keys_mem_size =  __n * sizeof(_KeyT);
+    const ::std::uint32_t __tmp_vals_mem_size =  __n * sizeof(_ValT);
+    const ::std::size_t __tmp_mem_size = __group_hists_mem_size + __global_hist_mem_size + __tmp_keys_mem_size + __tmp_vals_mem_size;
+
+    // ::std::uint8_t* __p_tmp_mem = sycl::malloc_device<::std::uint8_t>(__tmp_mem_size, __q);
```

Comment:
Is this comment required?

Reply:
Thanks for noticing it. It seems I forgot to replace `malloc_shared` with` malloc_device` after some debugging. The comment has to be uncommented:)